pym/bob/input.py

```diff
@@ -684,6 +746,57 @@ def isDeterministic(self):
     def hasJenkinsPlugin(self):
         return True
 
+    def callSubversion(self, workspacePath, *args):
+        cmdLine = ['svn']
+        cmdLine.extend(args)
+
+        try:
+            output = subprocess.check_output(cmdLine, cwd=workspacePath,
+                universal_newlines=True, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            raise BuildError("svn error: '{}' '{}'".format(" ".join(cmdLine), e.output))
+        return output
+
+    def status(self, workspacePath, dir, verbose=0):
+        scmdir = os.path.join(workspacePath, dir)
+        if not os.path.exists(os.path.join(os.getcwd(), scmdir)):
+            return 'error'
+
+        for m in self.__modules:
+            if m['dir'] == dir:
+                break;
+
+        ret = 0
+        svnoutput = self.callSubversion(os.path.join(os.getcwd(), workspacePath, m['dir']), 'status')
+        first = True
+        if len(svnoutput):
+            ret = printSCMStatus("    ! has modifications:", scmdir, verbose, ret)
+            if verbose >= 2:
+                for line in svnoutput.split('\n'):
+                    print('      '+line.rstrip())
+
+        _url = re.compile(r"^URL: {}")
+        _revision = re.compile(r"^Revision:")
+        svnoutput = self.callSubversion(os.path.join(os.getcwd(), workspacePath, m['dir']), 'info')
+        for line in svnoutput.split('\n'):
+            u = re.search('^URL: (.*)', line)
```

Comment:
This is dangerous. The output format is dependent on the locale. You should use the `--xml` option of subversion and get the information from there.